### ISSUE TITLE
DataMatrix barcodes support

### ISSUE THREAD

**Issue by lentyaishe** [tjwoon/csZBar#99]
Hi,

This barcode scanner plugin is much better than the official phonegap-barcode-scanner plugin. It's much more optimized and works really better and quicker. Also no crashes were witnessed unlike the official one.

Unfortunately there's a big BUT about it. It supports a very limited set of barcode types. Is it planned to add support to other types like DataMatrix?

Thanks

**Comment by uniphonic** [comment]
ZBar is the native library that the csZBar cordova plugin uses. There has been some talk about adding DataMatrix to ZBar, but it doesn't seem there is much done about it yet. 

https://sourceforge.net/p/zbar/feature-requests/31/

Because we need DataMatrix support as well, my company has gone back to the official phonegap-barcode-scanner plugin (which uses ZXing native library) on Android (which doesn't seem to suffer from the same stability issues of the iOS version, as far as I can tell), and on iOS we've begun to use this one, which instead of using open source libraries (ZBar or ZXing) it uses the iOS build in library AVFoundation: https://github.com/itraycer/XocializeScanner

I hope that someday someone will add DataMatrix support to ZBar.

**Comment by cleitonjar** [comment]
ZBar the best!

**Comment by lentyaishe** [comment]
Thanks for the reply!

Fortunately I've a [found a version](https://github.com/Telerik-Verified-Plugins/BarcodeScanner) of official plugin with the fix for the memory leak issue it has.

Good luck for anyone that needs this solution!

**Comment by uniphonic** [comment]
@lentyaishe Thanks for pointing out the Telerik version, but I just tried the [Telerik version](https://github.com/Telerik-Verified-Plugins/BarcodeScanner) my self, and unfortunately it still has the same error I was seeing in the [official version](https://github.com/phonegap/phonegap-plugin-barcodescanner), which is that on iOS after 80 or so scans, it stops recognizing all barcodes, until the app is restarted. This issue is reported here: https://github.com/phonegap/phonegap-plugin-barcodescanner/issues/370

Looks like I'm back to the https://github.com/itraycer/XocializeScanner for iOS. Thanks anyway!

**Comment by madscientist159** [comment]
Was very surprised to learn ZBar doesn't support data matrix.  Back to the handheld scanners for now, but would love to see this feature implemented!

**Comment by alexisdal** [comment]
yep... lack of DATA_MATRIX is a bit disappointing. I guess i'll have to find another way around too

**Comment by amijalis** [comment]
Just a bump — datamatrix support would be huge.

**Comment by disjustin** [comment]
In case anyone who ends up here is looking for datamatrix library: 
https://pypi.org/project/pylibdmtx/
https://github.com/NaturalHistoryMuseum/pylibdmtx
Maybe consider merging?
